### PR TITLE
Step 898 pod update

### PR DESCRIPTION
**Pull request by godrei**
### Checklist

- [x] I've read and accepted the [Contribution Guidelines](https://github.com/bitrise-steplib/.github/blob/main/CONTRIBUTING.md)
- Requires _MINOR_ [version update](https://semver.org/)

### Context

This PR adds an option for running `pod update` command, instead of the default `pod install`.

Resolves: https://github.com/bitrise-steplib/steps-cocoapods-install/issues/49

### Changes

- reorganise bitrise.yml wfs (local test wf, ci wf, e2e test case wfs)
- modify `test_with_gemfile` to test the `update` command: `test_update_with_gemfile`
- introduce `CocoapodsInstaller` and test the pod install phase
- introduce and wire in `command` input

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -13,63 +13,35 @@ import (
...
 )
 
 // ConfigsModel ...
 type ConfigsModel struct {
-	SourceRootPath  string
-	PodfilePath     string
-	Verbose         string
-	IsCacheDisabled string
+	Command         string `env:"command,opt[install,update]"`
+	SourceRootPath  string `env:"source_root_path,required"`
```
Comment (lpusok) [inline main.go:26]: It was checked to be a directory previously.
```suggestion
	SourceRootPath  string `env:"source_root_path,dir"`
```

File: main.go [hunk 2]
```diff
@@ -13,63 +13,35 @@ import (
...
+	PodfilePath     string `env:"podfile_path"`
```
Comment (lpusok) [inline main.go:27]: If provided, could be checked to be an existing file.

File: bitrise.yml [hunk 3]
```diff
@@ -78,14 +78,17 @@ workflows:
     - REPO_URL: https://github.com/bitrise-samples/ios-cocoapods-at-root.git
     - TEST_REPO_BRANCH: master
     - PODFILE_PTH: ''
+    - COMMAND: install
```
Comment (lpusok) [inline bitrise.yml:81]: Do not see it referenced. (`$COMMAND`)

**Review by lpusok — CHANGES_REQUESTED** [review]

File: main.go (hunk 2 above)
Comment (godrei) [reply main.go:27]: https://github.com/bitrise-steplib/steps-cocoapods-install/pull/68/files#diff-2873f79a86c0d8b3335cd7731b0ecf7dd4301eb19a82ef7a1cba7589b5252261R38

**Review by lpusok — APPROVED** [review]

**Pull request merged**